lib/readline.js

```diff
@@ -22,30 +22,21 @@ const stripVTControlCharacters = internalReadline.stripVTControlCharacters;
 
 
 exports.createInterface = function(input, output, completer, terminal) {
-  var rl;
-  if (arguments.length === 1) {
-    rl = new Interface(input);
-  } else {
-    rl = new Interface(input, output, completer, terminal);
-  }
-  return rl;
+  return new Interface(input, output, completer, terminal);
 };
 
 
 function Interface(input, output, completer, terminal) {
   if (!(this instanceof Interface)) {
-    // call the constructor preserving original number of arguments
-    const self = Object.create(Interface.prototype);
-    Interface.apply(self, arguments);
-    return self;
+    return new Interface(input, output, completer, terminal);
   }
 
   this._sawReturn = false;
 
   EventEmitter.call(this);
   var historySize;
 
-  if (arguments.length === 1) {
+  if (input && input.input) {
```

Comment:
In fact, given that `Interface` allows calling itself without `new` (and unless I'm missing something), it might be even reduced to `exports.createInterface = Interface;` (in which case I don't really see why two separate methods are exported, unless we want to move `Interface` to proper ES6 class and effectively forbid calling without `new` in future).